### PR TITLE
Update sds011.py

### PR DESCRIPTION
**Pull request by dolind**
sensor did not resume after setting to sleep and restarting program. Fixed this by waking up sensor and moving probe function

### CODE REVIEW

**Comment by dolind** [comment]
With this change i can use this module directly in a cron job, which collects one measurement. The sensor is woken up everytime the serial connection is made. After getting the measurement it can be set to sleep. Next time it will wake up.

At the first start the sensor is already woke up. This does not matter

**Comment by Erriez** [comment]
@dolind I'm curious, which platform are you using? I miss data regularly and the Python script may crash under Windows. This looks like related to the serial receive handling.

File: sds011/sds011.py [hunk 1]
```diff
@@ -148,6 +148,7 @@ def probe(self):
         return
 
     def get_sensor_data(self):
+        self.probe()
```
Comment (menschel) [inline sds011/sds011.py:151]: Actually probe() just looks for the sensor and updates the states variables. It makes no sense to do that on every data query.

File: sds011/sds011.py [hunk 2]
```diff
@@ -127,7 +127,7 @@ def __init__(self,port,modesel="active",rate=0,devid=0xFFFF,
         self.rx_queue_handler = threading.Thread(target = self.handle_rx)
         self.rx_queue_handler.setDaemon(True)
         self.rx_queue_handler.start()
-        self.probe()
+        self.wakeup()
```
Comment (menschel) [inline sds011/sds011.py:130]: probe() is an essential part of initialisation, it should not be removed from __init__()
wakeup() is only necessary if you have not set the working period correctly and/or set the sensor to sleep()

**Review by menschel — COMMENTED** [review]
There was a similar issue that the communication did not want to start properly.
I haven't found out if it was the sensor or handle_rx().
https://github.com/menschel/sds011/commit/1fa3b6e2b67f4478f565a884686d961a831435c6

**Comment by menschel** [comment]
To me this seems to be a unique setup with the CRON job. I acknowledge that it works but this should not be merged into a master.
If you set up the working period correctly, there should be no need for a CRON job.

**Pull request closed without merging**